src/zenml/models/component_model.py

```diff
@@ -53,10 +49,10 @@ class ComponentModel(AnalyticsTrackedModelMixin):
         "is_shared",
     ]
 
-    id: Optional[UUID] = Field(
-        default=None,
-        title="The id of the Stack Component.",
+    id: UUID = Field(
```

Comment:
Check out the Updates to the StackModel, specifically how it inherits from ShareableProjectScopedDomainModel - this might make handling our models easier if you also apply it here